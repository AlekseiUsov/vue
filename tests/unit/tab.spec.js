import { mount } from "@vue/test-utils";
import tab from "../../src/components/atoms/ToDoTab.vue";

describe("ToDoTab.vue", () => {
  it("тестируем отрисовку компонента", () => {
    const wrapper = mount(tab);
    expect(wrapper).toMatchSnapshot();
  });

  it("тестируем пропсы компонента", () => {
    const wrapper = mount(tab, {
      propsData: {
        name: "First post",
        isActive: true,
      },
    });
    expect(wrapper.find("li").text()).toBe("First post");
  });

  it("тестируем кнопку закрывания", async () => {
    const wrapper = mount(tab);

    await wrapper.find("li").trigger("click");
    expect(wrapper.emitted("changeTab")).toBeTruthy();
  });
});
