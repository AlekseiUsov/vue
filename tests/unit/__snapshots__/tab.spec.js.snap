// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoTab.vue тестируем отрисовку компонента 1`] = `
<li class="">

</li>
`;
